packages/medusa/src/api/routes/admin/product-categories/add-products-batch.ts

```diff
@@ -0,0 +1,116 @@
+import { IsArray, ValidateNested } from "class-validator"
+import { Request, Response } from "express"
+
+import { EntityManager } from "typeorm"
+import { ProductBatchProductCategory } from "../../../../types/product-category"
+import { ProductCategoryService } from "../../../../services"
+import { Type } from "class-transformer"
+import { FindParams } from "../../../../types/common"
+
+/**
+ * @oas [post] /product-categories/{id}/products/batch
+ * operationId: "PostProductCategoriesCategoryProductsBatch"
+ * summary: "Add Products to a category"
+ * description: "Assign a batch of products to a product category."
+ * x-authenticated: true
+ * parameters:
+ *   - (path) id=* {string} The ID of the Product Category.
+ * requestBody:
+ *   content:
+ *     application/json:
+ *       schema:
+ *         $ref: "#/components/schemas/AdminPostProductCategoriesCategoryProductsBatchReq"
+ * x-codegen:
+ *   method: addProducts
```

Comment:
todo(question): for sales channels and conditionResources, we have used `methodNameBatch`, does this will be the auto-generated method, or will it append the last segment of the route on it

Reply:
I think I can answer this: This will be the method name under the category resource, i.e. `productCategories.addProducts`. For sales channels we've done `addProducts` as well.

Reply:
<img width="882" alt="Screenshot 2023-01-27 at 10 04 26" src="https://user-images.githubusercontent.com/25098370/215048398-2d267204-8b1e-493c-82b4-8bd69e85fe31.png">
Sorry I meant those SC management 😂 but I believe that seb asked that batch was present in the naming. But I also think that there is no real convention on that @pKorsholm 

So feel free to ignore, it is just that at the moment depending on the batch route, the hook method are named differently which can loose the user in the end.

cc: @srindom

Reply:
al alright, yeah because this is the oas for sales channel batch adding: 
```
/**
 * @oas [post] /sales-channels/{id}/products/batch
 * operationId: "PostSalesChannelsChannelProductsBatch"
 * summary: "Add Products"
 * description: "Assign a batch of product to a sales channel."
 * x-authenticated: true
 * parameters:
 *   - (path) id=* {string} The ID of the Sales channel.
 * requestBody:
 *   content:
 *     application/json:
 *       schema:
 *         $ref: "#/components/schemas/AdminPostSalesChannelsChannelProductsBatchReq"
 * x-codegen:
 *   method: addProducts
```

Reply:
we might need to fix the convention on that as I think we are starting to mix then up

Reply:
Good shout to bring some consistency to this approach. I would keep this as is for now, and look at this holistically with other similar cases - sales channels, collections, etc. I don't want to create a deviation in convention here that will be lost.